src/aktualizr_secondary/aktualizr_secondary.h

```diff
@@ -3,62 +3,76 @@
 
 #include "aktualizr_secondary_config.h"
 #include "aktualizr_secondary_metadata.h"
-
-#include "aktualizr_secondary_interface.h"
-#include "msg_dispatcher.h"
+#include "msg_handler.h"
 
 #include "uptane/directorrepository.h"
 #include "uptane/imagerepository.h"
+#include "uptane/manifest.h"
 
 class UpdateAgent;
 class INvStorage;
 class KeyManager;
 
-class AktualizrSecondary : public IAktualizrSecondary {
+class AktualizrSecondary : public MsgDispatcher {
  public:
   using Ptr = std::shared_ptr<AktualizrSecondary>;
 
  public:
-  AktualizrSecondary(AktualizrSecondaryConfig config, std::shared_ptr<INvStorage> storage,
-                     std::shared_ptr<KeyManager> key_mngr, std::shared_ptr<UpdateAgent> update_agent);
-
-  std::tuple<Uptane::EcuSerial, Uptane::HardwareIdentifier, PublicKey> getInfo() const override;
-  Uptane::Manifest getManifest() const override;
-  bool putMetadata(const Uptane::RawMetaPack& meta_pack) override { return putMetadata(Metadata(meta_pack)); }
-  bool sendFirmware(const std::string& firmware) override;
-  data::ResultCode::Numeric install(const std::string& target_name) override;
-
-  MsgDispatcher& getDispatcher() { return dispatcher_; }
+  const Uptane::EcuSerial& serial() const { return ecu_serial_; }
+  const Uptane::HardwareIdentifier& hwID() const { return hardware_id_; }
+  PublicKey publicKey() const;
+  Uptane::Manifest getManifest() const;
+
+  virtual bool putMetadata(const Metadata& metadata);
+  virtual bool putMetadata(const Uptane::RawMetaPack& meta_pack) { return putMetadata(Metadata(meta_pack)); }
+
+  virtual data::ResultCode::Numeric install();
+  virtual void completeInstall() = 0;
+
+ protected:
+  AktualizrSecondary(AktualizrSecondaryConfig config, std::shared_ptr<INvStorage> storage);
+
+  // protected interface to be defined by child classes, i.e. a specific IP secondary type (e.g. OSTree, File, etc)
+  virtual bool getInstalledImageInfo(Uptane::InstalledImageInfo& installed_image_info) const = 0;
+  virtual bool isTargetSupported(const Uptane::Target& target) const = 0;
+  virtual data::ResultCode::Numeric installPendingTarget(const Uptane::Target& target) = 0;
+  virtual data::InstallationResult applyPendingInstall(const Uptane::Target& target) = 0;
+
+ protected:
```

Comment:
Redundant `protected`.

Reply:
Well, I did it for purpose. I grouped protected methods into two groups, the first one are pure virtual protected methods (kind of protected interface that must be implemented by a child class) and the second group is methods that are defined in this class and can be used by children. That's the style I was taught, i.g. group methods and fields and explicitly specify an access modifier for each group.  
Anyway, it's not a big deal, I can remove it and just leave a comment and additional whitespace between the groups.